docs/specification/current.md

```diff
@@ -240,45 +240,66 @@ A prefetch token is a placeholder in a prefetch template that is *_replaced by i
+
+For read, the prefetch token FHIRPath is limited to *only* the name of the context element - no further navigation within the context element is permitted.  For example, `{{context.draftOrders.id}}` is not a valid prefetch token for a read because it attempts to access the id field of the medication field.  If expanding the prefetch token does not result in a simple Resource/id string, the prefetch is in error and the prefetch cannot be returned.
+
+##### search prefetch
+The cue that a prefetch is a search is that it starts with a resource name followed by a '?'.  Search prefetches can be more complex than reads - and not all CDS clients will support all potential forms of search.  Additional considerations include:
+
+1. Prefetch templates can drill deeper into the context, navigating to child elements.  For example `{{context.draftOrders.id}}` or event `{{context.draftOrders.ofType(MedicationRequest).medicationReference.reference}}` would be valid tokens.
+2. Prefetch templates can resolve to an empty collection.  If the template resolves to no elements, the prefetch result will be __null__, regardless of where within the search expression the prefetch token falls.
+3. Prefetch templates can resolve to a collection of more than one element.  If this occurs, the collection is returned as single comma-delimited concatenated string.  E.g. `{{context.draftOrders.ofType(MedicationRequest).medicationReference.reference}}` might resolve to `Medication/123,http%3a//someserver.com/Medication/456,Medication/789`
+4. Prefetch templates can resolve to complex objects.  This is handled as follows:
+* If a collection element is a simple type, the value is returned and any extensions or 'id' elements are ignored.  If no value is specified for the element, the element is ignored
+* If a collection element is a Coding or Identifier, the element is converted to [token](http://hl7.org/fhir/search.html#token) syntax with code/value followed by '|' followed by system.
+5. Content expanded from a prefetch token may require HTTP escaping to ensure a valid URL fragment
+6. When dealing with prefetch tokens that resolve to relative references, in some cases, the search needs the resolved token to be an id or collection of ids rather than references. In this circumstance, the FHIRPath can be followed with ":" and the name of the resource whose references should be filtered.  Only relative references for the specified resource type are returned.  For example `{{context.draftOrders.ofType(MedicationRequest).requester.reference:Practitioner}}` would resolve to the a list of ids who are requesters that are also Practitioners.  For legacy reasons, this can also be done with the userId element by inserting the resource name into the context name.  E.g. `{{context.userPatientRoleId}}`, `{{context.userPractionerId}}`, `{{context.userPractitionerRoleId}}`, and `{{context.userRelatedPersonId}}`.
```

Comment:
This "extra colon at the end" syntax makes it harder to see what is the "thing that needs to be parsed as FHIRPath". I'd rather see (in order of preference):

* Define "simple FHIRpath + `replaceMatches`" so developers can build a list of IDs by taking final path segments
* Define a distinct templating syntax `{{expr}:Patient}` or somesuch so the fhirpath is unambiguous
* Let CDS Clients figure out how to filter based on context (i.e., "please do what I mean" -- if searching for )
* Use existing fhirpath comment syntax like `/*Patient*/` instead of tacking on `:`. Aim would be to leverage existing parsers. If you just split on `:` you hit complications in places where the gramamr (https://github.com/HL7/FHIRPath/blob/master/spec/fhirpath.g4) allows colons.